src/main/java/Duke.java

```diff
@@ -1,10 +1,221 @@
+        String BYE = "bye";
+        String LIST = "list";
+        String MARK = "mark";
+        String TODO = "todo";
+        String DEADLINE = "deadline";
+        String EVENT = "event";
+        String DELETE = "delete";
+
+        System.out.println(GREET);
+
+        ArrayList<String> instructionsList = new ArrayList<>();
+
+        Task task = new Task("hello");
+        Task[] tasks = new Task[100];
+        task.number = 0;
+
+        Deadline deadline = new Deadline("return book", "holi");
+        Event event = new Event("test", "7pm");
+
+        for (int i = 0; i < 200; i++) { // can have 200 input lines (including wrong command)
+            Scanner in = new Scanner(System.in);
+            //System.out.print("Type something: ");
+            task.instruction = in.nextLine();
+            String instructionLine = task.instruction.replaceAll("todo|deadline|event", "");
+            String updatedInstructionLine; //updated instruction line is in the form of [][] instructionline
+
+            Task t = new Task("read book");
+            String[] arrOfStr = task.instruction.split(" ", 50);
+            String[] arrOfDeadline = instructionLine.split("/by ", 2);
+            String[] arrOfEvent = instructionLine.split("/at ", 2);
+
+            boolean isBye = arrOfStr[0].equals(BYE); //if true then exit
+            boolean isList = arrOfStr[0].equals(LIST);
+            boolean isMark = arrOfStr[0].equals(MARK);
+            boolean isTodo = arrOfStr[0].equals(TODO);
+            boolean isDeadline = arrOfStr[0].equals(DEADLINE);
+            boolean isEvent = arrOfStr[0].equals(EVENT);
+            boolean isDelete = arrOfStr[0].equals(DELETE);
+            
+            String instructionNum;
+
+            try {
+
+                if (isBye) {
+
+                    System.out.println(EXIT);
+                    break;
+
+                } else if (isList) {
+
+                    System.out.println("Here are the task(s) in your list:");
+                    for (int j = 1; j <= task.number; j++) {
+                        System.out.println(j + ". " + instructionsList.get(j - 1));
+                    }
+
+                } else if (isMark) {
+
+                    if (arrOfStr.length == 1){
+                        throw new DukeException("Error: You have not entered the task number!");
+                    }
+
+                    instructionNum = arrOfStr[1];
+                    int index = Integer.parseInt(instructionNum) - 1;
+                    t.setStatusIcon(true);
+                    String temp = instructionsList.get(index);
+                    String prefix = "  \\[T]\\[ ]";
+                    temp = temp.replaceAll(prefix, "  [T][X]");
+                    instructionsList.set(index, temp); //updates the list
+                    System.out.println("Nice! I've marked this task as done:");
+                    System.out.println(instructionsList.get(index));
```

Comment:
Consider extracting this chunk into a new method for SLAP and better code readablility.

Same issue in other `else if` blocks as well.